### DISCUSSION TITLE
Resubmit a rejected invoice.

### DISCUSSION THREAD

**Discussion by BOTSolutions8686** [lavaloon-eg/ksa_compliance#138] [Q&A] [answered]
Hi . 
What is the way to resubmit an invoice that shows rejected in the ZATCA integration log list ? 

Thank you

**Comment by mhaggag** [comment, marked as answer]
Hello.

In the `Sales Invoice Additional Fields` list, you should find the latest rejected record (status = Rejected with the `Is Latest` flag checked):
![image](https://github.com/user-attachments/assets/5e6faad0-1fc7-4616-900c-e194b39e1782)

Open it and you'll find a button at the top to `Fix Rejection`:
![image](https://github.com/user-attachments/assets/c2e9defb-ceca-4034-a501-13181341bac6)

This creates a new `Sales Invoice Additional Fields` document for the same invoice using your current configuration. This is meant to address cases where the rejection was the result of a bad configuration (in which case you fix the rejection after updating configuration) or an application bug (in which case you do it after updating the app to fix the bug).

**Reply by BOTSolutions8686** [reply]
Understood . 
Thank you for the detailed explanation. I understand now.
This is a very smart approach . We are in the process of separating the submitting to zatca action from the submit invoice action. (Something that should be considered for the module itself) 
as we were using another module prior to this, we have some invoice in the system that are submitted in the erp but not actually sent to zatca yet. This case could be true for other people. As many people will be shifting to your module as your module is much more stable and provides much more functionality. 
It could be added as part of the configuration in the module 
"Send invoice on submit" if check the module will work as it is working now. If uncheck the module will add a separate button on the invoice form "Send to ZATCA"

**Reply by mhaggag** [reply]
Thanks for the feedback. Please let us know if you run into issues with the fix rejection feature or if you have ideas for improvement. Right now it's a conservative feature in that it addresses limited cases, hence why there's no batch fix rejection--we want to understand the scenarios first.

For the separation of invoice submission from submission to ZATCA, we have support for two modes: Live mode where an invoice is immediately submitted to ZATCA after submission, and a batch mode where we submit hourly. In `ZATCA Business Settings`, look for the "Sync with ZATCA" dropdown:
![image](https://github.com/user-attachments/assets/f112a88e-6ed9-4100-beba-db94b1f92744)

Note that regardless of the mode, the `Sales Invoice Additional Fields` document will be created for the invoice immediately after submission (on_submit hook) to ensure proper ordering of the invoice counter and previous invoice hash submitted to ZATCA. In batch mode, it'll sit in status `Ready For Batch` until the next hourly job run.

It's possible to run the batch job early and with a start date from a custom page titled "Einvoicing Sync":
![image](https://github.com/user-attachments/assets/aae092be-4c77-4e9b-9f59-e934a7fa8e90)

This is a minimal page that allows you to select a date and queue a job to submit all `Sales Invoice Additional Fields` documents with status = `Ready For Batch` or `Resend` issued since that date. It was created before the hourly sync job, and remains more of a recovery tool in case there are issues with the job.

I don't think we ever considered a manual mode, since it increases the likelihood of human error. @omarashrafsarhan can advise if it's been or is being considered in the roadmap.

**Reply by BOTSolutions8686** [reply]
Thank you very much for the detailed explanation . 
I understand that the bulk and live feature is for b2b(Live) and b2c(Bulk) respectively. But can be used as desired. 
The manual mode can be considered for the use case as follows
I currently have an invoice in the system, 023458 It is an approved invoice (Document status is submitted) but it is not cleared by ZATCA. 
At the moment I am unable to find a way to submit this particular invoice to zatca.
And I have 15 invoices like this one.

**Reply by omarashrafsarhan** [reply]
In the app you can send the invoices based on the status of the additional fields we didn't want any human interfering to prevent sending repeated invoices, however you can use "Einvoicing sync" page to force sending the invoices before the sync background job to send all the invoices with "Resend" or "Ready for batch".
The only status that requires human interfering is the rejected and based on the rejection reason, the user must correct it and send it from the sales invoice additional field record.
My question for you @BOTSolutions8686 why you need the manual sync on the invoice level while you can use the sync page, I just need more details of the objective of this case so we can consider it in our roadmap

**Reply by BOTSolutions8686** [reply]
Sync page can only be used for batch submissions - where as in the ZATCA settings we have set it to be live for B2b operations - More suitable. 

Lets say 
My customer in this case have invoice in the system from 1st October 2024 that needs to be submitted to ZATCA. 
The document status at the moment is "Submitted" 
Second scenario 
We changed from an old module to your module. during this transition my customer created 1 invoice which is approved internally "Document status approved" Zatca status "NOT SENT"  - My question is how can I send this invoice. This one single invoice to zatca for clearance. 

As for the human error - Previously we had custom ZATCA Status apart from document status on the invoice. If the ZATCA Status was = cleared - the Sent to ZATCA button will not appear on the invoice from anymore.  (In the way customers did not had to go to the integration log and was able to see right there if an invoice is cleared or not. if not cleared - adjust "Sent to ZATCA" )

**Reply by mhaggag** [reply]
Regarding transitioning from another app to this one: Our recommendation would be to sync all current invoices using the previous app, and to not issue any new invoices during the transition (uninstall the old app, then install the new one and onboard).

The human error with manual invoice submission is that users might submit some invoices to ZATCA but not others, which puts them at risk if they are audited. My understanding after discussing it with the business team is that any gaps in submitted invoice serials, for example, makes it appear as if it's tax evasion. They've disabled invoice cancellation for similar reasons. Since I'm more on the technical side of things, @omarashrafsarhan , @khaledYusufEG or @Ahmed-M-Fawzy can elaborate. Perhaps it'd be better to create a different discussion for this topic specifically (manual submission of invoices)

Regarding invoice status: I agree that it's currently not ideal that you can't see an invoice's ZATCA status directly from the invoice. We generally try to avoid modifying standard doctypes, especially really busy ones like sales invoice. We'll look into adding a virtual status field or something similar. @omarashrafsarhan, we should create a work item for this if we don't have one already.

**Reply by BOTSolutions8686** [reply]
Thank you again for you detailed reponse . Spent two days trying to bring the ZATCA Status from Additional invoice fields to invoice using a custom field and a form script. 
The Form script did work but then it added and update button on top of the invoice every time the status got submitted. which is not ideal . 
The status did show up on the invoice form using the form script . but it seems impossible to add it to the list. 

I intended to reply here with a solution and the form script I wrote but I won't as the solution is still not ideal. 
I will wait for you to include this in the module core features. "As we are not cut from the same cloth of developers , my skills seems to be more Chinese :-D "
